spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/MapTaskExecutionDao.java

```diff
@@ -203,6 +203,15 @@ public Long getTaskExecutionIdByJobExecutionId(long jobExecutionId) {
 		}
 	}
 
+	@Override
+	public void updateExternalExecutionId(long taskExecutionId, String externalExecutionId) {
+		if(!this.taskExecutions.containsKey(taskExecutionId)) {
+			throw new IllegalStateException("Invalid TaskExecution, ID " + taskExecutionId + " not found.");
+		}
+		TaskExecution taskExecution = taskExecutions.get(taskExecutionId);
```

Comment:
This can be simplified / made more robust (threadsafety) by asserting that `get` did not return `null`